cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -482,10 +482,10 @@ public Void call() throws Exception {
           context.setState(new ProgramState(ProgramStatus.INITIALIZING, null));
           if (mapReduce instanceof ProgramLifecycle) {
             ((ProgramLifecycle) mapReduce).initialize(context);
+            ((AbstractMapReduce) mapReduce).initialize();
```

Comment:
The `MapReduceRuntimeService` shouldn't be aware of the existence of `AbstractMapReduce`.